packages/react-data-grid-react-window/src/components/DataGrid/useDataGridStyles.styles.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  DataGridState,
+  makeStyles,
+  mergeClasses,
+  useDataGridStyles_unstable as useDataGridStylesBase_unstable,
+} from '@fluentui/react-components';
+
+const useStyles = makeStyles({
+  root: {
+    // DataGrid gets min-width: fit-content applied directly to the element, thus the need for !important
+    // without auto width, the DataGrid will not scroll
+    minWidth: 'auto !important',
```

Comment:
instead of `!important` styles that are applied in all cases, how about just override the inline style from the component's state in the resizable columns scenario?

```ts
export const useDataGrid_unstable = (
  props: DataGridProps,
  ref: React.Ref<HTMLElement>
): DataGridState => {
  const { targetDocument } = useFluent();
  const scrollbarWidth = useScrollbarWidth({ targetDocument });

  let containerWidthOffset = props.containerWidthOffset;

  if (containerWidthOffset === undefined) {
    containerWidthOffset = props.selectionMode
      ? -TABLE_SELECTION_CELL_WIDTH
      : 0;
    containerWidthOffset -= scrollbarWidth || 0;
  }

  const baseState = useBaseState(
    { ...props, 'aria-rowcount': props.items.length, containerWidthOffset },
    ref
  );

  if (props.resizableColumns && baseState.root.style) {
    baseState.root.style.minWidth = 'auto';
  }

  return baseState;
};

```

Reply:
yep that makes much more sense, a lot better than needing `!important` - updated